### PR TITLE
DBZ-1317 Collection id is optional in MongoDB events

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.jboss.org/browse/DBZ-1317

@jpechane, that's a fallout from the source info refactoring.

### CODE REVIEW

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSourceInfoStructMaker.java [hunk 1]
```diff
@@ -35,9 +35,14 @@ public Schema schema() {
     @Override
     public Struct struct(SourceInfo sourceInfo) {
         final Struct result = super.commonStruct(sourceInfo)
-                .put(SourceInfo.REPLICA_SET_NAME, sourceInfo.collectionId().replicaSetName())
-                .put(SourceInfo.COLLECTION, sourceInfo.collectionId().name())
-                .put(SourceInfo.ORDER, sourceInfo.position().getInc())
+                .put(SourceInfo.REPLICA_SET_NAME, sourceInfo.replicaSetName());
+
+        if (sourceInfo.collectionId() != null) {
+            result.put(SourceInfo.COLLECTION, sourceInfo.collectionId().name());
```
Comment (jpechane) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSourceInfoStructMaker.java]: @gunnarmorling I think schema needs to be updated as collection is mandatory.

**Comment by jpechane** [comment]
@gunnarmorling Would it be possible to extend `io.debezium.connector.mongodb.RecordMakersTest` (or similar) with no-op event test?

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSourceInfoStructMaker.java (hunk 1 above)
Comment (gunnarmorling) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSourceInfoStructMaker.java]: Is it really optional, though? IIUC, the recorded position in the oplog (offset) might denote an event that's not specific to a collection, but would the connector ever expose a CDC event which doesn't apply to a collection?

Comment (jpechane) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSourceInfoStructMaker.java]: @gunnarmorling The question is - either no-op does not come to this code - and then there is noo need for fix or it does and then it is optional.

Comment (gunnarmorling) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSourceInfoStructMaker.java]: I think the root problem is one we had before: insufficient split of the logic for updating the source info based on the latest event received from the oplog and then "rendering" the source info struct. IIUC, the latter should only happen if we have a collection id at this point. I'll do some more digging.

**Comment by gunnarmorling** [comment]
@jpechane reworked this a bit and cleaned things up on a more general level now. There's `AbstractSourceInfo#struct()` now, which is the only method rendering the SI into a `Struct`. Interactions in the Mongo one are still a bit convoluted, but that'll become better once Mongo is moved to `OffsetContext`, in which case its `SourceInfo` will be free from any attributes solely relevant for the offset but not the `source` struct.

**Comment by jpechane** [comment]
@gunnarmorling LGTM now, thanks! Would it be possible to add the noop test as the last step?

**Pull request merged**